pyk/src/pyk/kcfg/minimize.py

```diff
@@ -188,6 +200,87 @@ def _fold_lift(result: bool, finder_lifter: tuple[Callable, Callable]) -> bool:
+
+        Specifically, when `B_merge = B_x or B_y or ... or B_z`
+        - `or`: fresh variables in places where the configurations differ
+        - `Edge` in A_merged -|Edge|-> B_merge: list of merged edges is from A_i -|Edge|-> B_i
+        - `Split` in B_merge -|Split|-> B_x, B_y, ..., B_z: subst for it is from A -|Split|-> A_1, A_2, ..., A_n
+        :param semantics: provides the is_mergeable heuristic
+        :return: whether any merge was performed
+        """
+
+        def _is_mergeable(x: KCFG.Edge | KCFG.MergedEdge, y: KCFG.Edge | KCFG.MergedEdge) -> bool:
+            return self.semantics.is_mergeable(x.target.cterm, y.target.cterm)
+
+        # ---- Match ----
+
+        # A -|Split|> Ai -|Edge/MergedEdge|> Mergeable Bi
+        sub_graphs: list[tuple[KCFG.Split, list[list[KCFG.Edge | KCFG.MergedEdge]]]] = []
+
+        for split in self.kcfg.splits():
+            _edges = [
+                single(self.kcfg.general_edges(source_id=ai))
+                for ai in split.target_ids
+                if self.kcfg.general_edges(source_id=ai)
+            ]
+            _partitions = [p for p in partition(_edges, _is_mergeable) if len(p) > 1]
+            if _partitions:
+                sub_graphs.append((split, _partitions))
+
+        # ---- Rewrite ----
+
+        merged = False
+        for split, edge_partitions in sub_graphs:
+            merged = True
+
+            # Remove the original sub-graphs
+            for p in edge_partitions:
+                for e in p:
+                    self.kcfg.remove_node(e.source.id)
+
+            # Create A -|MergedEdge|-> Merged_Bi -|Split|-> Bi, if one edge partition covers all the splits
+            if len(edge_partitions) == 1 and len(edge_partitions[0]) == len(split.target_ids):
+                merged_bi_cterm, merged_bi_subst = cterms_anti_unify(
+                    [edge.target.cterm for edge in edge_partitions[0]], True, self.kdef
+                )
+                merged_bi = self.kcfg.create_node(merged_bi_cterm)
+                self.kcfg.create_merged_edge(split.source.id, merged_bi.id, edge_partitions[0])
+                self.kcfg.create_split(
+                    merged_bi.id, zip([e.target.id for e in edge_partitions[0]], merged_bi_subst, strict=True)
+                )
+                continue
+
+            # Create A -|Split|-> Others & Merged_Ai -|MergedEdge|-> Merged_Bi -|Split|-> Bi
+            _split_nodes = [t for t in split.target_ids if t not in {e.source.id for p in edge_partitions for e in p}]
+            for edge_partition in edge_partitions:
+                merged_ai_cterm, _ = cterms_anti_unify(
+                    [ai2bi.source.cterm for ai2bi in edge_partition], True, self.kdef
+                )
+                merged_bi_cterm, merged_bi_subst = cterms_anti_unify(
+                    [ai2bi.target.cterm for ai2bi in edge_partition], True, self.kdef
+                )
```

Comment:
```suggestion
                merged_ai_cterm, _ = cterms_anti_unify(
                    [ai2bi.source.cterm for ai2bi in edge_partition], keep_values=True, kdef=self.kdef
                )
                merged_bi_cterm, merged_bi_subst = cterms_anti_unify(
                    [ai2bi.target.cterm for ai2bi in edge_partition], keep_values=True, kdef=self.kdef
                )
```

Make sure to keep the named parameters' names.